src/local_device.rs

```diff
@@ -691,7 +706,7 @@ fn load_recovery_device(key_file: PathBuf, password: String) -> FutureIntoCorout
     FutureIntoCoroutine::from(async move {
         platform_device_loader::load_recovery_device(&key_file, &password)
             .await
-            .map(LocalDevice)
+            .map(|local_device| LocalDevice(Arc::new(local_device)))
```

Comment:
You can add `.map(Arc::new)` before instead

Reply:
done